iso15118/secc/states/iso15118_2_states.py

```diff
@@ -970,8 +970,29 @@ async def process_message(
             )
 
             response_code = ResponseCode.OK
-            if current_authorization_status.certificate_response_status:
-                response_code = current_authorization_status.certificate_response_status
+            if resp_status := current_authorization_status.certificate_response_status:
+                # according to table 112 of ISO 15118-2, the Response code
+                # for this message can only be one of the following:
+                # OK, FAILED,
+                # FAILED_SEQUENCE_ERROR, FAILED_SIGNATURE_ERROR,
+                # FAILED_UNKNOWN_SESSION, FAILED_CHALLENGE_INVALID
+                # FAILED_CERTIFICATE_EXPIRED, FAILED_CERTIFICATE_REVOKED,
+                # FAILED_NO_CERTIFICATE_AVAILABLE
+                response_code = (
+                    resp_status
+                    if resp_status
+                    in [
+                        ResponseCode.OK,
+                        ResponseCode.FAILED,
+                        ResponseCode.FAILED_SEQUENCE_ERROR,
+                        ResponseCode.FAILED_SIGNATURE_ERROR,
+                        ResponseCode.FAILED_UNKNOWN_SESSION,
+                        ResponseCode.FAILED_CERTIFICATE_EXPIRED,
+                        ResponseCode.FAILED_CERTIFICATE_REVOKED,
+                        ResponseCode.FAILED_NO_CERTIFICATE_AVAILABLE,
+                    ]
+                    else ResponseCode.FAILED
+                )
```

Comment:
This could potentially change in an upcoming update to the spec based on the ticket here: https://iso15118.elaad.io/pt2/15118-2/user-group/-/issues/38
Also André has opened a ticket for a better response code when chain error occurs here: https://iso15118.elaad.io/pt2/15118-2/user-group/-/issues/41